packages/next/pages/_document.tsx

```diff
@@ -765,7 +765,10 @@ export class Head extends Component<HeadProps> {
             {!hasCanonicalRel && (
               <link
                 rel="canonical"
-                href={canonicalBase + cleanAmpPath(dangerousAsPath)}
+                href={
+                  canonicalBase +
+                  require('../server/utils').cleanAmpPath(dangerousAsPath)
+                }
               />
             )}
```

Comment:
Correct me if I was wrong, but I thought `hasCanonicalRel` is runtime evaluated, thus it wouldn't strip `cleanAmpPath` from the bundle.

Reply:
We have `process.env.NEXT_RUNTIME !== 'edge' && inAmpMode` a couple of lines above, that will remove this dependency for the edge runtime.